Add org_mozilla_vrbrowser.clients_last_seen and org_mozilla_vrbrowser…

scholtzan opened the pull request:
Addresses: #497

scholtzan commented on sql/org_mozilla_vrbrowser_derived/clients_daily_v1/query.sql:
```diff
@@ -0,0 +1,169 @@
+    client_id,
+    sample_id,
+    ROW_NUMBER() OVER w1_unframed AS _n,
+    'Firefox Reality' AS app_name,
```
Is there a list with these names somewhere?

jklukas replied:
There's a partial list in https://docs.telemetry.mozilla.org/concepts/choosing_a_dataset_mobile.html#products-overview but it's kind of a judgment call right now.

We should definitely have a more coherent and well-documented story for this as part of generalizing derived tables for Glean apps.

scholtzan commented on sql/org_mozilla_vrbrowser_derived/clients_daily_v1/query.sql:
```diff
@@ -0,0 +1,169 @@
+    NULL AS baseline_metrics,
+    metrics
+  FROM
+    org_mozilla_vrbrowser_stable.metrics_v1
```
I looked into using `LEFT JOIN`, the best I could come up with but also doesn't quite work was:
```diff
   FROM
     org_mozilla_vrbrowser_stable.baseline_v1
   UNION ALL
+  -- Consider users only as active if they sent a baseline ping.
   SELECT
-    submission_timestamp,
-    document_id,
-    client_info,
-    sample_id,
-    metadata,
-    normalized_channel,
+    metrics.submission_timestamp,
+    metrics.document_id,
+    metrics.client_info,
+    metrics.sample_id,
+    metrics.metadata,
+    metrics.normalized_channel,
     NULL AS baseline_metrics,
-    metrics
+    metrics.metrics
   FROM
-    org_mozilla_vrbrowser_stable.metrics_v1
+    org_mozilla_vrbrowser_stable.baseline_v1 AS baseline
+  LEFT JOIN
+    org_mozilla_vrbrowser_stable.metrics_v1 AS metrics
+  ON
+    baseline.client_info.client_id = metrics.client_info.client_id
+  WHERE
+    baseline.client_info.client_id IS NOT NULL
 ),
   --
 base AS (
```

jklukas replied:
Good point. I think we'll need to do separate aggregations on the baseline and metrics pings and then do the join between the two CTEs that each have only one row per client.

This also touches on a broader issue that we'd like to be able to support new glean applications automatically for GUD, etc. The baseline ping is standard for all apps and is intended as the source of being able to calculate retention, etc. If we need to aggregate baseline and metrics separately anyway to do the join here, maybe the best way forward here is to define a completely generic `baseline_daily` query and then individual apps can define custom `metrics_daily` queries that can be joined in as needed.

So, if it seems like doing separate aggregations and then joining is a straight-forward solution here, let's do that for now. Otherwise, we can table this for the moment and discuss the more general way forward for glean aggregations during Berlin and include @fbertsch in the discussion.

scholtzan replied:
I we do separate aggregations would we simply use dimensions related to `client_info` from `baseline_daily`? That behaviour (and potentially event results?) would be different from using `udf_mode_last` on both baseline and metrics pings.

jklukas replied:
We would used `udf_mode_last` over baseline pings for the fields that exist in baseline pings, and we would use `udf_mode_last` over metrics pings for any metrics-ping-specific fields.

Yes, that could lead to us choosing a different value for a dimension compared to using `udf_mode_last` over baseline and metrics pings. But it seems conceptually reasonable to me to accept the value we see from the baseline ping. Baseline pings become the main source of truth, and metrics pings are considered only for enriching the data with dimensions that exist as metrics.

scholtzan commented:
I added `baseline_daily` and `metrics_daily` and updated `clients_daily` and `clients_last_seen`. If that seems reasonable I'll also add `metadata.yaml` files for the queries.

scholtzan commented:
I added metadata files. I think the table descriptions could use some improvements

scholtzan commented on templates/org_mozilla_vrbrowser_derived/metrics_daily_v1/metadata.yaml:
```diff
@@ -0,0 +1,9 @@
+  A daily aggregate of metrics pings from each Firefox Reality client,
+  partitioned by day
+owners:
+  - ascholtz@mozilla.com
```
not sure if that should be me?

jklukas replied:
If you feel comfortable with being listed as owner, go for it. But feel free to mark `jklukas@mozilla.com` as the owner, or as an additional owner.

scholtzan replied:
good idea, I added you as owner

scholtzan commented:
> We'll need to run the queries manually to create the tables in order to get the dry-run test passing. Let me know if you'd like to coordinate on that.

I'd be doing that some time today. We could coordinate over zoom whenever you have time today.
